### ISSUE TITLE
Adafactor requires a "Max grad norm" argument

### ISSUE THREAD

**Issue by RookHyena** [bmaltais/kohya_ss#2153]
The optimizer "Adafactor" expects an input of 0 in the "Max grad norm" argument. I noticed this when LoRA training. 

Without that input, you will get the following warning:
"WARNING  because max_grad_norm is set, clip_grad_norm is enabled. consider set to 0 / max_grad_normが設定されているためclip_grad_normが有効になります。0に設定して無効にしたほうがいいかもしれません        train_util.py:3807"
The English warning is a bit ambiguous, but the Japanese warning states that it would be better to disable clip_grad_norm by setting max_grad_norm to 0. 

The following is the printed training command I am using:
`accelerate launch --num_cpu_threads_per_process=2 "./train_network.py" --bucket_no_upscale --bucket_reso_steps=1 --cache_latents --cache_latents_to_disk --caption_extension=".txt" --learning_rate="4e-07" --logging_dir="Z:\Stable_Diffusion_Training\SD_Scripts\AI_Dev\log" --lr_scheduler="constant_with_warmup" --lr_scheduler_num_cycles="10" --lr_warmup_steps="25304" --max_data_loader_n_workers="0" --resolution="512,512" --max_token_length=225 --max_train_steps="316300" --min_snr_gamma=10 --mixed_precision="bf16" --network_alpha="64" --network_args conv_dim="64" conv_alpha="64" --network_dim=64 --network_module=networks.lora --network_train_unet_only --noise_offset=0.0357 --optimizer_args scale_parameter=False relative_step=False warmup_init=False --optimizer_type="Adafactor" --output_dir="Z:\Stable_Diffusion_Training\SD_Scripts\AI_Dev\model" --output_name="DevV2" --pretrained_model_name_or_path="runwayml/stable-diffusion-v1-5" --save_every_n_epochs="1" --save_model_as=safetensors --save_precision="fp16" --save_state --seed="1234" --train_batch_size="1" --train_data_dir="Z:\Stable_Diffusion_Training\SD_Scripts\AI_Dev\img" --unet_lr=4e-07 --xformers --sample_sampler=euler_a --sample_prompts="Z:\Stable_Diffusion_Training\SD_Scripts\AI_Dev\model\sample\prompt.txt" --sample_every_n_epochs="1"`

What is missing from the training command is `--max_grad_norm=0`, however, the GUI only generates a `--max_grad_norm` argument when the input is a non-zero number. For example, when the "Max grad norm" input is 0, the `--max_grad_norm=0` argument is not generated. When the "Max grad norm" input is 1, the `--max_grad_norm=1` argument is generated.

I am unsure if the other optimizer requires a `--max_grad_norm=0` argument. However, the 'train_util.py' only has this warning when the optimizer "Adafactor" is in use, line 3806 is nested in the "Adafactor" argument decomposition. Also the 'train_util.py' seems to indicate on line 2746 that `--max_grad_norm` is set to 1.0 by default. So it might be better to exclude the `--max_grad_norm` generation only for an input of 1 and not 0. Though it also might be better to always generate the `--max_grad_norm` argument.

**Comment by bmaltais** [comment]
Hi, thank you for pointing this out. I will fix the GUI to address this. You can test the update in the dev branch code.

**Comment by RookHyena** [comment]
Awesome! I will close the issue.
